File: articles/why-analytical-psychology-still-matters/go/relevance_score.go
package main

import "fmt"

func relevanceScore(symbolicDepth, meaningCoherence, clinicalUtility, culturalInterpretivePower, revisionCapacity, doctrinalRigidity float64) float64 {
	return 0.62*symbolicDepth +
		0.58*meaningCoherence +
		0.54*clinicalUtility +
		0.48*culturalInterpretivePower +
		0.60*revisionCapacity -
		0.70*doctrinalRigidity
}

func main() {
	score := relevanceScore(0.82, 0.74, 0.58, 0.77, 0.69, 0.21)
	fmt.Printf("Synthetic relevance score: %.3f\n", score)
}

File: articles/why-analytical-psychology-still-matters/fortran/relevance_score.f90
! Why Analytical Psychology Still Matters
! Fortran example: simple conceptual relevance score

program relevance_score_demo
  implicit none

  real :: symbolic_depth, meaning_coherence, clinical_utility
  real :: cultural_interpretive_power, revision_capacity, doctrinal_rigidity
  real :: score

  symbolic_depth = 0.82
  meaning_coherence = 0.74
  clinical_utility = 0.58
  cultural_interpretive_power = 0.77
  revision_capacity = 0.69
  doctrinal_rigidity = 0.21

  score = 0.62 * symbolic_depth + 0.58 * meaning_coherence + &
          0.54 * clinical_utility + 0.48 * cultural_interpretive_power + &
          0.60 * revision_capacity - 0.70 * doctrinal_rigidity

  print *, "Synthetic relevance score:", score
end program relevance_score_demo
